### PR TITLE
Create feeds for other honeypot types (source ips)

### PR DESCRIPTION
**Pull request by MayADevBe**
Resolving pre-commit-hook changes
Source IP extraction for 'general' honeypots
Add 'generl' honeypot list to IOC
Base search extend to include differently named time column Add Gernal Honeypots (tested)

# Description

### Main Changes
- added 'general' honeypot source IP extraction
  - added list to IOC with the honeypots that IOC was seen in
  - changed all relevant parts in the code - including tests+doc
  - constant 'GENERAL_HONEYPOTS' includes names of all honeypots that will be included (allows for specification, what honeypots should be included)
  - API queries are done with all lowercase, while Elasticsearch is done case sensitive
  - did not change cowrie/log4j, because of additional extraction + change of existing data sets

**Tested Honeypots:**
"Heralding","Ciscoasa","Honeytrap","Dionaea","ConPot","Adbhoney","Tanner","CitrixHoneypot","Mailoney","Ipphoney","Ddospot","ElasticPot","Dicompot","Redishoneypot","Sentrypeer"
- I did not have data to test other honeypots, but in principle, other honeypots should work the same way (might need adjustment of base search - see beneath)

- adjusted base search ('greedybear\cronjobs\base.py')
  - some honeypots log do not have the 'timestamp' column, so I added a 'should' query, where one of the different columns should exist
  - this might be further refined depending on the name of the columns with the timestamps of untested honeypots

### Smaller Changes
- changed the 'flake8' repo from GitLab to GitHub link (they seemed to have moved)
- added to Installation Doc how to create a superuser

### Examples
API query: "/api/feeds/heralding/all/recent.txt"
Result:
```
# These feeds are generated by The Honeynet Project once every 10 minutes and are protected by the following license: https://github.com/honeynet/GreedyBear/blob/main/FEEDS_LICENSE.md
192.168.82.1
```

API query: `/api/enrichment?query=192.168.82.1`
Result:
```
{
"found":true,
"ioc":{
    "id":1,
    "name":"192.168.82.1",
    "type":"ip",
    "first_seen":"2022-12-07T03:50:00.456653",
    "last_seen":"2022-12-07T06:40:09.655173",
    "days_seen":["2022-12-07"],
    "number_of_days_seen":1,
    "times_seen":235,
    "log4j":false,
    "cowrie":true,
    "general":["heralding","ciscoasa","honeytrap","dionaea","conpot","adbhoney","tanner","citrixhoneypot","mailoney","ipphoney","ddospot","elasticpot","dicompot"],
    "scanner":true,
    "payload_request":false,
    "related_ioc":[]
    },
"query":"192.168.82.1"
}
```

## Related issues

#10  

## Type of change

- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://greedybear.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [x] I have added documentation of the new features.
- [x] The tests gave 0 errors.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://greedybear.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [x] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
- [x] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](./Contribute.md)).
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

### CODE REVIEW

File: greedybear/cronjobs/base.py [hunk 1]
```diff
@@ -31,9 +31,19 @@ def _base_search(self, honeypot):
         search = Search(using=self.elastic_client, index="logstash-*")
         self.log.debug(f"minutes_back_to_lookup: {self.minutes_back_to_lookup}")
         gte_date = f"now-{self.minutes_back_to_lookup}m/m"
-        search = search.filter(
-            "range", **{"timestamp": {"gte": gte_date, "lte": "now/m"}}
+        # Some honeypots have different column for the time
```
Comment (mlodic) [inline greedybear/cronjobs/base.py:34]: cool, did you test it with a local T-POT?

File: tests/greedybear/cronjobs/test_cowrie.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from greedybear.models import IOC
 
 
-class Log4PotTestCase(TestCase):
+class CowrieTestCase(TestCase):
```
Comment (mlodic) [inline tests/greedybear/cronjobs/test_cowrie.py:10]: you also found my bad copy/paste 🤣

**Review by mlodic — APPROVED** [review]
Thank you for your PR!

I really loved how much dedication you had in this PR and how you tried to explain everything you have done! This is extremely mature and useful for the maintainers. Thank you very much for this! The PR is extremely well done for a first contributor!

File: greedybear/cronjobs/base.py (hunk 1 above)
Comment (MayADevBe) [reply greedybear/cronjobs/base.py:34]: Yes, tested it on a local VM with Greedybear on the same machine

**Pull request merged**